### ISSUE TITLE
Select all frames, pls?

### ISSUE THREAD

**Issue by kubapl2004gmail** [TheWilley/Fruity_Dance_Sprite_Sheet_Generator#24]
Select all frames button, pls?

**Comment by TheWilley** [comment]
Hi! Thanks for the suggestion!

That's a good idea, there can be cases where one would want to edit multiple cells at the same time. I've actually never considered this until now, so thanks for bringing this to my attention!

The underlying system to handle selections is flawed, so first and foremost I would have to do some changes there. Secondly, because I never designed the system to handle multiple selections I need to make some changes there too. So this might not make it to the next version, but I'll try to add it to the one after that (although no promises).

I imagine this will work as it does in excel, where you ctrl + click to pick many cells. Alternatively, you just click many cells, and If you click one that already has been selected it will be unselected.

I'll work on this when I got some time.

**Comment by TheWilley** [comment]
I have decided to implement this sooner, as it was not as hard as I thought. It will be in the next release, at least the multiple selection part.

As for the "select all button" it is yet to be implemented. This will come later probably, as I want to make sure multiple selections work properly first.

**Comment by TheWilley** [comment]
Short update on this feature: It will be implemented in v2.6.0, rather than v2.5.0. This is because selections work trough mouse clicks, and I'm unsure as of now on how to do it another way. I will look into this tho, as its a handy feature.

**Comment by TheWilley** [comment]
This initially seemed like a challenge, but apparently it could be solved with `8` lines! In fact, here is is:
```typescript
public selectAll() {
	// Step 1, get all image elements
	const allImageElements = document.querySelectorAll<HTMLElement>(".dropzones img");
	// Step 2, check if all elements are selected
	for (const element of allImageElements) {
		this.show_controls(element);
	}
}
```

I am suprised thing worked out this well, as I've been overthinking the solution all the time. In any case, this issue is solved by d01d62268156220e9e5cea185375b6e330aac8cf
